src/verify/VerificationStartScreen.tsx

```diff
@@ -122,9 +123,20 @@ function VerificationStartScreen({
           />
         ),
       headerLeft: () => route.params?.hideOnboardingStep && <BackButton />,
+      // Disable iOS back during onboarding
+      gestureEnabled: route.params?.hideOnboardingStep ? false : true,
     })
   }, [navigation, step, totalSteps, route.params])
 
+  // Prevent device back on Android during onboarding
+  useEffect(() => {
+    if (!route.params?.hideOnboardingStep) {
+      const backPressListener = () => true
+      BackHandler.addEventListener('hardwareBackPress', backPressListener)
+      return () => BackHandler.removeEventListener('hardwareBackPress', backPressListener)
+    }
+  }, [])
```

Comment:
Can we use `navigation.addListener('beforeRemove', ...` instead to handle both iOS and Android with the same code?

Reply:
We'd still have to have the `gestureEnabled` even with the `navigation.addListener('beforeRemove', ...`, [docs](https://reactnavigation.org/docs/preventing-going-back/).
> this feature also doesn't work properly with [@react-navigation/native-stack](https://reactnavigation.org/docs/native-stack-navigator). To make this work, you need to:
> - Disable the swipe gesture for the screen (gestureEnabled: false)...